File: dotnet/src/SemanticKernel.Abstractions/Memory/DataEntryBase.cs
// Copyright (c) Microsoft. All rights reserved.

using System;
using System.Text.Json.Serialization;

namespace Microsoft.SemanticKernel.Memory;

/// <summary>
/// A base class for data entries.
/// </summary>
public class DataEntryBase
{
    /// <summary>
    /// Creates an instance of a <see cref="DataEntryBase"/>.
    /// </summary>
    /// <param name="key">The data key.</param>
    /// <param name="timestamp">The data timestamp.</param>
    [JsonConstructor]
    public DataEntryBase(string? key = null, DateTimeOffset? timestamp = null)
    {
        this.Key = key ?? string.Empty;
        this.Timestamp = timestamp;
    }

    /// <summary>
    /// Gets the key of the data.
    /// </summary>
    [JsonPropertyName("key")]
    public string Key { get; set; }

    /// <summary>
    /// Gets the timestamp of the data.
    /// </summary>
    [JsonPropertyName("timestamp")]
    public DateTimeOffset? Timestamp { get; set; }

    /// <summary>
    /// <c>true</c> if the data has a timestamp.
    /// </summary>
    [JsonIgnore]
    public bool HasTimestamp => this.Timestamp.HasValue;
}

File: java/semantickernel-api/src/main/ignore/java/com/microsoft/semantickernel/memory/DataEntryBase.java
// Copyright (c) Microsoft. All rights reserved.
package com.microsoft.semantickernel.memory;

import java.time.ZonedDateTime;
import javax.annotation.Nonnull;
import javax.annotation.Nullable;

/** A base class for data entries. */
public class DataEntryBase {
    @Nonnull private final String key;
    @Nullable private final ZonedDateTime timestamp;

    /**
     * Creates an instance of a {@code DataEntryBase}.
     *
     * @param key The key of the data.
     * @param timestamp The timestamp of the data.
     */
    public DataEntryBase(@Nullable String key, @Nullable ZonedDateTime timestamp) {
        this.key = key != null ? key : "";
        this.timestamp = timestamp;
    }

    /**
     * Gets the key of the data.
     *
     * @return The key of the data
     */
    public String getKey() {
        return key;
    }

    /**
     * Gets the timestamp of the data.
     *
     * @return The timestamp of the data.
     */
    @Nullable
    public ZonedDateTime getTimestamp() {
        return timestamp;
    }
}
